Samples/BackgroundBlur/BackgroundBlur/TransformAsync.cpp

```diff
@@ -0,0 +1,1212 @@
+        INFINITE);      // indefinite wait
+   
+
+    switch (dwWaitResult) {
+    case WAIT_OBJECT_0:
+        // TODO: Capture time and write to preview
+        if (g_now == NULL){
+            g_now = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+            //OutputDebugString(L"First time responding to event!");
+        }
+        else {
+            auto l_now = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+            auto timePassed = l_now - g_now;
+            g_now = l_now;
+            auto fps = 30000 / timePassed; // TODO: marco on num frames to update after? 
+            OutputDebugString(L"THREAD: ");
+            OutputDebugString(std::to_wstring(fps).c_str());
+            OutputDebugString(L"\n");
+
+            auto message = std::wstring(L"Frame Rate: ") + std::to_wstring(fps) + L" FPS";
+            MainWindow::_SetStatusText(message.c_str());
+            //TRACE(("Responded to event and it's been %d miliseconds", timePassed));
+            // TODO: Call Set status text with new framerate
+        }
+
+        break;
+    default: 
+        TRACE(("Wait error (%d)\n", GetLastError()));
+        return 0;
+    }
+    return 1;
+}
+
+HRESULT TransformAsync::OnSetD3DManager(ULONG_PTR ulParam)
+{
+    HRESULT hr = S_OK;
+    com_ptr<IUnknown> pUnk;
+    com_ptr<IMFDXGIDeviceManager> pDXGIDeviceManager;
+    com_ptr<ID3D11DeviceContext> pContext;
+    com_ptr<ID3D11Device5> pDevice;
+    if (ulParam != NULL) {
+        pUnk.attach((IUnknown*)ulParam);
+        pDXGIDeviceManager = pUnk.try_as<IMFDXGIDeviceManager>();
+        if (pDXGIDeviceManager && m_spDeviceManager != pDXGIDeviceManager) {
+            // Invalidate dx11 resources
+            InvalidateDX11Resources();
+
+            // Update dx11 device and resources
+            m_spDeviceManager = pDXGIDeviceManager;
+            UpdateDX11Device();
+
+            // Hand off the new device to the sample allocator, if it exists
+            if (m_spOutputSampleAllocator)
+            {
+                CHECK_HR(hr = m_spOutputSampleAllocator->SetDirectXManager(pUnk.get()));
+            }
+
+            // Create event and thread for framerate
+            m_hFenceEvent = CreateEvent(NULL,               // Security attributes
+                FALSE,              // Reset token to false means system will auto-reset event object
+                FALSE,              // Initial state is nonsignaled
+                TEXT("FrameEvent"));  // Event object name
+            
+            DWORD dwThreadID;
+        }
+    }
+    else
+    {
+        // Invalidate dx11 resources
+        InvalidateDX11Resources();
+    }
+
+done:
+    return hr;
+}
+
+// TODO: Clean up this function
+HRESULT TransformAsync::UpdateDX11Device()
+{
+    HRESULT hr = S_OK;
+    com_ptr<ID3D11DeviceContext> pContext;
```

Comment:
>pContext [](http://example.com/codeflow?start=33&length=8)

looks like mixing name coding between sp and p prefixes to smart ptrs.  Generally I believe the guidance is to remove for external code 

[https://www.osgwiki.com/wiki/Cpp_Core_Guidelines#NL.5_Don.27t_encode_type_information_in_names](https://www.osgwiki.com/wiki/Cpp_Core_Guidelines#NL.5_Don.27t_encode_type_information_in_names)